runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -844,58 +844,65 @@ Java_jdk_internal_misc_Unsafe_objectFieldOffset1(JNIEnv *env, jobject receiver,
 		}
 	}
 	vmFuncs->internalExitVMToJNI(currentThread);
-	
+
 	return offset;
 }
 
 /**
  * Writes modified memory in an address range from cache to main memory.
  * Uses memory barriers before and after writeback to ensure ordering.
- * 
+ *
  * On x86, the writeback is done using the CLWB instruction if available
  * for performance, falling back to CLFLUSHOPT then CLFLUSH otherwise.
- * 
+ *
  * @param addr address of block to write back to memory
  * @param len length of block being written
  */
 void JNICALL
 Java_jdk_internal_misc_Unsafe_writebackMemory(JNIEnv *env, jobject receiver, jlong addr, jlong len)
 {
 /* Exclude Windows since it does not support GCC assembly syntax,
- * and Linux is the only target actually specified in JEP 352
+ * and Linux is the only target actually specified in JEP 352.
  */
-#if ((defined(J9X86) || defined(J9HAMMER)) && !defined(WIN32))
-	J9VMThread *currentThread = (J9VMThread*)env;
+#if (defined(J9X86) || defined(J9HAMMER)) && !defined(WIN32)
+	J9VMThread *currentThread = (J9VMThread *)env;
 	J9JavaVM *vm = currentThread->javaVM;
 	uintptr_t cacheLineSize = vm->dCacheLineSize;
 
 	if (cacheLineSize > 0) {
-		U_8 *endAddr = (*(U_8 **) &addr) + (*(uintptr_t *) &len) - 1;
-		U_8 *ptr = (U_8 *)((*(uintptr_t *) &addr) & ~(cacheLineSize - 1));
+		uintptr_t const uaddr = (uintptr_t)(U_64)addr;
+		uintptr_t const ulen = (uintptr_t)(U_64)len;
+		uintptr_t const firstCacheLine = uaddr & ~(cacheLineSize - 1);
+		uintptr_t const lastCacheLine = (uaddr + ((0 == ulen) ? 0 : (ulen - 1))) & ~(cacheLineSize - 1);
+
+		/* adjust for pre-increment in the loops below */
+		uintptr_t cacheLine = firstCacheLine - cacheLineSize;
 
 		VM_AtomicSupport::readWriteBarrier();
-		switch(vm->cpuCacheWritebackCapabilities) {
-			case J9PORT_X86_FEATURE_CLWB:
-				for (; ptr < endAddr; ptr += cacheLineSize) {
-					asm volatile("clwb %0" : "+m" (*ptr));
-				}
-				/* writeback any partial cache line at the end */
-				asm volatile("clwb %0" : "+m" (*endAddr));
-				break;
-			case J9PORT_X86_FEATURE_CLFLUSHOPT:
-				for (; ptr < endAddr; ptr += cacheLineSize) {
-					asm volatile("clflushopt %0" : "+m" (*ptr));
-				}
-				asm volatile("clflushopt %0" : "+m" (*endAddr));
-				break;
-			case J9PORT_X86_FEATURE_CLFSH:
-				for (; ptr < endAddr; ptr += cacheLineSize) {
-					asm volatile("clflush %0" : "+m" (*ptr));
-				}
-				asm volatile("clflush %0" : "+m" (*endAddr));
-				break;
-			default:
-				goto error;
+		switch (vm->cpuCacheWritebackCapabilities) {
+		case J9PORT_X86_FEATURE_CLWB:
+			do {
+				cacheLine += cacheLineSize;
+				asm volatile("clwb %0" : "+m" (*(U_8 *)cacheLine));
+			}
```

Comment:
Please put the whiles on the brace line.

Reply:
Fixed.